main.c

```diff
@@ -34,11 +34,16 @@ static void die_on(const bool condition, const char *string, ...)
     }
 }
 
+// Data provided by the VMM to the loader via the fw_cfg device.
 struct boot_params {
-    uintptr_t kernel_addr;
-    uintptr_t initrd_addr;
-    uintptr_t initrd_size_addr;
-    uintptr_t cmdline_addr;
+    // Physical address of the kernel image.
+    const uintptr_t kernel_addr;
+    // Physical address of the initrd image.
+    const uintptr_t initrd_addr;
+    // Physical address of the initrd size.
+    const uintptr_t initrd_size_addr;
+    // Physical address of the cmd line string (a null-terminated C-string).
+    const uintptr_t cmdline_addr;
```

Comment:
if all these fields should be constant, why not have a const instance of `boot_params`?

Reply:
It is: `static void linux_boot(const struct boot_params boot_params)`

I have never written large C projects with others, so I do not know what modern code styles are. From my intuition, I mark everything as const because I think it is better style, isn't it?

Reply:
Generally yes, but while you are designing a data structure, you remove the choice from the user (maybe some test implementer or something you can't foresee) to handle the content as they wish.
While at the same time, regardless of the individual fields being const or not, the user can still always make the whole data structure const.

Reply:
thanks!